test/receiveAndDeleteMode.spec.ts

```diff
@@ -0,0 +1,368 @@
+    throw new Error("Define QUEUE_NAME in your environment before running integration tests.");
+  }
+  if (!process.env.SUBSCRIPTION_NAME) {
+    throw new Error(
+      "Define SUBSCRIPTION_NAME in your environment before running integration tests."
+    );
+  }
+
+  namespace = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
+  queueClient = namespace.createQueueClient(process.env.QUEUE_NAME, {
+    receiveMode: ReceiveMode.receiveAndDelete
+  });
+  topicClient = namespace.createTopicClient(process.env.TOPIC_NAME);
+  subscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.SUBSCRIPTION_NAME,
+    {
+      receiveMode: ReceiveMode.receiveAndDelete
+    }
+  );
+  deadletterQueueClient = namespace.createQueueClient(
+    Namespace.getDeadLetterQueuePathForQueue(queueClient.name)
+  );
+  deadletterSubscriptionClient = namespace.createSubscriptionClient(
+    Namespace.getDeadLetterSubcriptionPathForSubcription(
+      topicClient.name,
+      subscriptionClient.subscriptionName
+    ),
+    subscriptionClient.subscriptionName
+  );
+
+  const peekedQueueMsg = await queueClient.peek();
+  if (peekedQueueMsg.length) {
+    throw new Error("Please use an empty queue for integration testing");
+  }
+
+  const peekedSubscriptionMsg = await subscriptionClient.peek();
+  if (peekedSubscriptionMsg.length) {
+    throw new Error("Please use an empty Subscription for integration testing");
+  }
+  errorWasThrown = false;
+}
+
+async function afterEachTest(): Promise<void> {
+  await namespace.close();
+}
+
+describe("ReceiveBatch from Queue/Subscription", () => {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function sendReceiveMsg(
+    senderClient: QueueClient | TopicClient,
+    receiverClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+    await senderClient.send(testMessages[0]);
+    const msgs = await receiverClient.receiveBatch(1);
+
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    should.equal(msgs[0].body, testMessages[0].body);
+    should.equal(msgs[0].messageId, testMessages[0].messageId);
+    should.equal(msgs[0].deliveryCount, 0);
+  }
+
+  async function testNosettlment(
+    senderClient: QueueClient | TopicClient,
+    receiverClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+    await sendReceiveMsg(senderClient, receiverClient);
+    await testPeekMsgsLength(receiverClient, 0);
+  }
+
+  it("Queue: No settlement of the message removes message", async function(): Promise<void> {
+    await testNosettlment(queueClient, queueClient);
+  });
+
+  it("Subscription: No settlement of the message removes message", async function(): Promise<void> {
+    await testNosettlment(topicClient, subscriptionClient);
+  });
+});
+
+describe("Streaming Receiver from Queue/Subscription", function(): void {
```

Comment:
The tests inside this block needs to be repeated with autoComplete disabled.
Pass `autoComplete` as a variable to `sendReceiveMsg`, then you can re-use that code